node-config-docsgen/action.yml

```diff
@@ -0,0 +1,159 @@
+name: "Generate Configuration Documentation"
+description: "Generate and upload Stacks Core configuration documentation"
+branding:
+  icon: "file-text"
+  color: "blue"
+
+inputs:
+  output:
+    description: "Output file for generated documentation"
+    required: false
+    default: "./node-parameters.md"
+  min-doc-size:
+    description: "Minimum documentation size in bytes"
+    required: false
+    default: "50000"
+  rust-nightly-version:
+    description: "Specific nightly version to use (e.g., 'nightly-2025-06-17')"
+    required: false
+    default: "nightly-2025-06-17"
+
+runs:
+  using: "composite"
+  steps:
+    ## Checkout the code
+    - name: Checkout the latest code
+      id: git_checkout
+      uses: actions/checkout@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
+
+    ## Install required system dependencies
+    - name: Install system dependencies
+      id: install_deps
+      shell: bash
+      run: |
+        echo "Installing system dependencies..."
+        sudo apt-get update -q
+        sudo apt-get install -y jq
+
+    ## Install Rust nightly toolchain (required for rustdoc JSON output)
+    - name: Install Rust nightly toolchain
+      id: install_rust
+      uses: dtolnay/rust-toolchain@b3b07ba8b418998c39fb20f53e8b695cdcc8de1b
```

Comment:
for consistency, this is the worfklow we use elsewhere for rust: https://github.com/actions-rust-lang/setup-rust-toolchain

ex: https://github.com/stacks-network/actions/blob/main/stacks-core/cache/build-cache/action.yml#L47-L51